infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationClientConfigListener.java

```diff
@@ -138,9 +141,22 @@ private void handleRegistrationTableEntries(List<CorfuStreamEntry> registrationT
                         .setClientName(clientName).setModel(model).build();
                 // TODO (V2): Currently we add a default subscriber for logical group use case instead of listening
                 //  on client registration. Subscriber should be added upon registration after grpc stream for session
-                //  creation is added.
-                // configManager.onNewClientRegister(subscriber);
-                // sessionManager.createOutgoingSessionsBySubscriber(subscriber);
+                if (model.equals(ReplicationModel.LOGICAL_GROUPS)) {
+                    for (LogReplicationSession session : sessionManager.getOutgoingSessions()) {
+                        CorfuMessage.HeaderMsg.Builder header = CorfuMessage.HeaderMsg.newBuilder()
+                                .setSession(session)
+                                .setVersion(getDefaultProtocolVersionMsg())
+                                .setIgnoreClusterId(true)
+                                .setIgnoreEpoch(true);
+                        LogReplication.LogReplicationSinkSessionInitializationMsg msg =
+                                LogReplication.LogReplicationSinkSessionInitializationMsg.newBuilder().setSession(session).build();
+                        CorfuMessage.RequestPayloadMsg payload =
+                                CorfuMessage.RequestPayloadMsg.newBuilder().setLrSinkSessionInitialization(msg).build();
+                        sessionManager.getRouter().getServerChannelAdapter().send(getRequestMsg(header.build(), payload));
+                    }
+                }
+                configManager.onNewClientRegister(subscriber);
+                sessionManager.createOutgoingSessionsBySubscriber(subscriber);
```

Comment:
the router must also be updated with the new session info, which seems to be missing.